listener.go

```diff
@@ -74,6 +74,21 @@ func (l *listener) Accept() (tpt.CapableConn, error) {
 			sess.CloseWithError(errorCodeConnectionGating, "connection gated")
 			continue
 		}
+
+		// return through active hole punching if any
+		key := sess.RemoteAddr().String()
+		var wasHolePunch bool
+		l.transport.holePunchingMx.Lock()
+		holePunch, ok := l.transport.holePunching[key]
+		if ok && !holePunch.fulfilled {
+			holePunch.connCh <- conn
+			wasHolePunch = true
+			l.transport.holePunching[key].fulfilled = true
```

Comment:
We're storing these by-pointer, right? `holePunch.fulfilled = true`?

Reply:
Yeah, I had to change that. Can't assign to a struct in a map without initialising a new struct.